src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -971,27 +960,15 @@ int xcldev::device::runTestCase(const std::string& exe,
         return -EINVAL;
     }
 
-    if (m_idx != 0)
-        idxOption = "-d " + std::to_string(m_idx);
-
-    std::string cmd = "";
-    if (isPython) {
-        cmd = "python " + exePath + " -k " + xclbinPath + " " + idxOption;
-    } else {
-        cmd = exePath + " " + xclbinPath + " " + idxOption;
-    }
+    std::string cmd = "python " + xrtTestCasePath + " -k " + xclbinPath + " -d " + std::to_string(m_idx) + " 2>&1"; //catch stderr
```

Comment:
2>&1 is bash specific? does it work with any shell, like csh?

Reply:
I just tried it on csh shell, both passing and failing scenarios. It works fine.

Reply:
> 
> 
> I just tried it on csh shell, both passing and failing scenarios. It works fine.

what do you get if you manually run this cmd string in csh?

Reply:
It doesn't work when I run the command manually... According to "popen" doc, the command is passed to /bin/sh using the -c flag

Reply:
do we want to explicitly specify bash in the cmd string? /bin/sh can point to any shell...

Reply:
No we don't

Reply:
so what if /bin/sh is pointing to csh?

Reply:
Oooh. Ok I see your point. This won't work in that case. I'll change this